### ISSUE TITLE
The Upload Failed Error using ADAM in Apps

### ISSUE THREAD

**Issue by slicksystems** [2sic/2sxc#1431]
**I'm submitting a ...**  
[x] bug report

**...about**   
[x] edit experience / UI
[x] admin experience UI
[x] Content Types or data management

**Current behavior**
I'm trying to create a simple App that contains a set of documents. The Content-Type I'm having trouble with has 2 fields: 1. Title (String, default) 2. Docs (Hyperlink, library).  When I try to upload, an error is produced.

When trying to upload the documents I'm getting the error:

_"The upload failed. The most likely cause is that the file you were trying to upload is bigger than the maximum upload size."_

![image](https://user-images.githubusercontent.com/13353102/33918872-03dc3ae0-df73-11e7-929d-cf78fe16b839.png)

![image](https://user-images.githubusercontent.com/13353102/33905120-d212174c-df3a-11e7-90d6-c082fa16f3ab.png)

**Expected behavior**

When I create Content, and not an App, following the same structure, the files upload without issue.

Here you can see in simple content, the files have uploaded as expected:

![image](https://user-images.githubusercontent.com/13353102/33896505-3594258c-df20-11e7-9cd4-fd1dad71f6cb.png)

**Minimal reproduction of the problem with instructions**

1. Fresh DNN 8 or 9 Install
2. Create new APP
3. Add hyperlink library field to the app
4. Try to Insert a new record by selecting 1 or more files with the ADAM uploader
5. You receive the error after trying to upload.

**Please tell us about your environment:**

* **2sxc version(s):** 9.1.12  
* **Browser:** [all]
* **DNN:** [all | 8&9 (most current) ] 
* **Language:** [English]

**Comment by slicksystems** [comment]
![image](https://user-images.githubusercontent.com/13353102/33905567-4798e454-df3c-11e7-815f-54eed9077473.png)

AbsoluteURL:/desktopmodules/2sxc/api/app-content/65b45d0e-f7f5-42c2-ad48-34bdbf8a93c5/7ea94f82-935b-42cd-a0bc-14320612bb03/Docs
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:f98d2b80-c8c3-44bb-bc62-8fb72e8745fe
AssemblyVersion:
PortalId:-1
UserId:-1
TabId:-1
RawUrl:
Referrer:
UserAgent:
ExceptionHash:GPJm4r5at1QhFXs60NmtYN87GI0=
Message:Object reference not set to an instance of an object.
StackTrace:
at ToSic.SexyContent.Adam.AdamController.UploadOne(String contentTypeName, Guid guid, String field, String subFolder, Boolean usePortalRoot) in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\Sxc WebApi\Adam\AdamController.cs:line 62 at ToSic.SexyContent.Adam.AdamController.Upload(String contentType, Guid guid, String field, String subFolder, Boolean usePortalRoot) in C:\Projects\2SexyContent\Web\DesktopModules\ToSIC_SexyContent\Sxc WebApi\Adam\AdamController.cs:line 47 at lambda_method(Closure , Object , Object[] ) at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass10.b__9(Object instance, Object[] methodParameters) at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync(HttpControllerContext controllerContext, IDictionary`2 arguments, CancellationToken cancellationToken) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Http.Tracing.ITraceWriterExtensions.d__18`1.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() at System.Web.Http.Controllers.ApiControllerActionInvoker.d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Http.Tracing.ITraceWriterExtensions.d__18`1.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() at System.Web.Http.Filters.ActionFilterAttribute.d__5.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Web.Http.Filters.ActionFilterAttribute.d__5.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Http.Filters.ActionFilterAttribute.d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() at System.Web.Http.Controllers.ActionFilterResult.d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Http.Filters.AuthorizationFilterAttribute.d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Http.Filters.AuthorizationFilterAttribute.d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Http.Filters.AuthorizationFilterAttribute.d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Http.Filters.AuthorizationFilterAttribute.d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() at System.Web.Http.Controllers.ExceptionFilterResult.d__0.MoveNext()
InnerMessage:
InnerStackTrace:
Source:ToSic.SexyContent.WebApi

**Comment by slicksystems** [comment]
Here are the logs...

[errorlog.zip](https://github.com/2sic/2sxc/files/1553210/errorlog.zip)

**Comment by iJungleboy** [comment]
I'm not sure if this is still an Issue. Especially since nobody else reported it. Could you install 9.32-RC and see if it still occurs? If yes, please also check the internal logging (there's a new cool detailed API-Logging in 9.32) to see what the real cause is - check out this blog to help you: https://2sxc.org/en/blog/post/using-2sxc-insights

My guess is it's probably an extension-filter (pdf, jpg, etc.) that was set or something, and _Insights_ should tell you that. 

I'll assume it's fixed, if not, post here and I'll reopen it.

**Comment by MaxiAnton** [comment]
Hello there,

I ran in an similiar issue today. 
![2023-06-12 13_36_24-Window](https://github.com/2sic/2sxc/assets/15210721/ce7be3da-69dc-42cb-bd44-807dad0b84df)

The Upload seems to work but it wont display.
If I close the UI Element for an Upload and reopen it, I can count my latest number of my files up until I see it BUT I still cant use it.
I already checked the user rights. Nothing noticeable.
There are also no problems in my dev console or in my DNN logs.
